eq-author/src/App/QuestionnairesPage/QuestionnairesTable/index.js

```diff
@@ -38,120 +43,163 @@ const Panel = styled.div`
+
+  const [state, dispatch] = usePersistedReducer(
+    STORAGE_KEY,
+    STORED_KEYS,
+    reducer,
+    { currentPageIndex: 0 },
+    setQuestionnaires(questionnairesRef.current)
+  );
+
+  if (questionnaires !== questionnairesRef.current) {
+    dispatch({
+      type: "SET_QUESTIONNAIRES",
+      payload: questionnaires,
+    });
+    questionnairesRef.current = questionnaires;
+  }
 
-  renderRows = questionnaires =>
-    questionnaires.map(questionnaire => {
-      return (
-        <Row
-          key={questionnaire.id}
-          autoFocus={questionnaire.id === this.state.focusedId}
-          questionnaire={questionnaire}
-          onDeleteQuestionnaire={this.handleOpenDeleteQuestionnaireDialog}
-          onDuplicateQuestionnaire={this.props.onDuplicateQuestionnaire}
-        />
-      );
+  const handleDeleteQuestionnaire = questionnaire => {
+    dispatch({
+      type: "DELETE_QUESTIONNAIRE",
+      payload: questionnaire,
     });
+    onDeleteQuestionnaire(questionnaire.id);
+  };
 
-  render() {
-    const { questionnaires } = this.props;
-    const { showDeleteQuestionnaireDialog, deleteQuestionnaire } = this.state;
-
-    return (
-      <Panel>
-        <Table>
-          <thead>
-            <tr>
-              <TH colWidth="50%">Title</TH>
-              <TH colWidth="15%">Created</TH>
-              <TH colWidth="20%">Owner</TH>
-              <TH colWidth="15%">Actions</TH>
-            </tr>
-          </thead>
-          <TransitionGroup component="tbody">
-            {this.renderRows(questionnaires)}
-          </TransitionGroup>
-        </Table>
-        <DeleteConfirmDialog
-          isOpen={showDeleteQuestionnaireDialog}
-          onClose={this.handleCloseDeleteQuestionnaireDialog}
-          onDelete={this.handleDeleteQuestionnaire}
-          title={deleteQuestionnaire && deleteQuestionnaire.displayName}
-          alertText="This questionnaire including all sections and questions will be deleted."
-          icon={questionConfirmationIcon}
-          data-test="delete-questionnaire"
+  return (
+    <Panel>
+      <Table>
+        <thead>
+          <tr>
+            <TH colWidth="50%">Title</TH>
+            <TH colWidth="15%">Created</TH>
+            <TH colWidth="20%">Owner</TH>
+            <TH colWidth="15%">Actions</TH>
+          </tr>
+        </thead>
+        <TransitionGroup component="tbody">
+          {state.currentPage.map((questionnaire, index) => {
+            return (
+              <Row
+                key={questionnaire.id}
+                autoFocus={questionnaire.id === state.autoFocusId}
+                questionnaire={questionnaire}
+                onDeleteQuestionnaire={handleDeleteQuestionnaire}
+                onDuplicateQuestionnaire={onDuplicateQuestionnaire}
+                isLastOnPage={state.currentPage.length === index + 1}
+              />
+            );
+          })}
+        </TransitionGroup>
+      </Table>
+      <TableFooter>
```

Comment:
I noticed on [the prototype](https://prototype-sharing--author-app.netlify.com/#/), the footer with the pagination and results text is displayed outside the table. Should this be moved to outside the panel?

Reply:
Yeah, as discussed requires a minor refactor to get this working. But you are correct and I'm making the change now